lib/web_ui/lib/src/engine/view_embedder/embedding_strategy/custom_element_embedding_strategy.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/src/engine/dom.dart';
+
+import 'embedding_strategy.dart';
+
+class CustomElementEmbeddingStrategy extends EmbeddingStrategy {
+  CustomElementEmbeddingStrategy(this._hostElement) {
+    // Clear children...
+    while (_hostElement.firstChild != null) {
+      _hostElement.removeChild(_hostElement.lastChild!);
+    }
+  }
+
+  final DomElement _hostElement;
+
+  @override
+  void initialize({
+    Map<String, String>? embedderMetadata,
+  }) {
+    // ignore:avoid_function_literals_in_foreach_calls
+    embedderMetadata?.entries.forEach((MapEntry<String, String> entry) {
+      _setHostAttribute(entry.key, entry.value);
+    });
+    _setHostAttribute('flt-embedding', 'custom-element');
+  }
+
+  @override
+  void attachGlassPane(DomElement glassPaneElement) {
+    glassPaneElement
+      ..style.width = '100%'
+      ..style.height = '100%'
+      ..style.display = 'block'
+      ..style.overflow = 'hidden'
+      ..style.position = 'relative';
```

Comment:
> Could there be a difference?

Yes, if we set `position: absolute` here, we're forcing users to make the container they provide `position: relative`, which they may (or may not) want to do.

If we make this `position: relative`, everything inside will Work As Expected, and the user shouldn't need to do anything extra for their flutter app to work.

Reply:
I'm curious about setting this through the inset rather than width/height. It might affect the div not having browser layout or something, but I'm giving it a shot anyways!

Reply:
Ah, `display: block` is required to ensure the glass-pane covers the whole app (and mouse events work), otherwise by default it's `display: inline`:

<img width="291" alt="Screen Shot 2022-12-14 at 4 31 12 PM" src="https://user-images.githubusercontent.com/1255594/207744600-aeb3cd0e-d1cc-4ff3-8d54-f5e8250c5b77.png">

`inset: 0px` needs `position: absolute` to work as intended, otherwise it's not enough to completely size the element  (but it has the problem I described above).